cadc-pkg-server/src/main/java/org/opencadc/pkg/server/PackageItem.java

```diff
@@ -73,42 +73,119 @@
      */
-    public PackageItem(URL url, String relativePath) {
-        if (url == null) {
-            throw new IllegalArgumentException("parameter url required.");
+    public PackageItem(String relativePath) {
+        if (!StringUtil.hasText(relativePath)) {
+            throw new IllegalArgumentException("relativePath is null or empty");
         }
+        this.relativePath = relativePath;
+        this.linkTarget = null;
+        this.content = null;
+    }
 
+    /**
+     * Creates a resource for a package. The relative path is used to create
+     * the file structure inside a package.
+     *
+     * @param relativePath path to the resource in the package.
+     * @param content URL to the file.
+     */
+    public PackageItem(String relativePath, URL content) {
         if (!StringUtil.hasText(relativePath)) {
-            throw new IllegalArgumentException("parameter relativePath required.");
+            throw new IllegalArgumentException("relativePath is null or empty");
+        }
+        if (content == null) {
+            throw new IllegalArgumentException("content is null");
         }
+        this.relativePath = relativePath;
+        this.content = content;
+        this.linkTarget = null;
+    }
 
-        this.url = url;
+    /**
+     * Creates a resource for a package. The relative path is used to create
+     * the file structure inside a package.
+     *
+     * @param relativePath path to the resource in the package.
+     * @param linkTarget path to the resource in the package.
+     */
+    public PackageItem(String relativePath, String linkTarget) {
+        if (!StringUtil.hasText(relativePath)) {
+            throw new IllegalArgumentException("relativePath is null or empty");
+        }
+        if (!StringUtil.hasText(relativePath)) {
+            throw new IllegalArgumentException("linkTarget is null or empty");
+        }
         this.relativePath = relativePath;
+        this.linkTarget = linkTarget;
+        this.content = null;
+    }
+
+    /**
+     * Get the relative path of the item (directory, file, or symbolic link) within the package archive.
+     *
+     * @return relative path to the item.
+     */
+    public String getRelativePath() {
+        return this.relativePath;
     }
 
+    /**
+     * Get the relative path within the package archive for the target of a symbolic link item.
+     *
+     * @return relative path of the symbolic link target, null if the item is a file or directory.
+     */
+    public String getLinkTarget() {
+        return this.linkTarget;
+    }
 
-    public URL getURL() {
-        return url;
+    /**
+     * Get the URL to the content of a file item.
+     *
+     * @return URL to the content for a file, null if the item is a directory or symbolic link.
+     */
+    public URL getContent() {
+        return this.content;
     }
 
-    public String getRelativePath() {
-        return relativePath;
+
```

Comment:
relativePath is never null via ctor checks,  so those checks below are not needed

otherwise, isFile only needs `content != null` and isSymbolicLink only needs `linkTarget != null`

style: I prefer to use the `this.` qualifier when assigning to a member variable (LHS of =), but not when using it. It makes it more clear when "state" is being modified.

Reply:
Updated for first two comments. I think I've heard the third before and will try to better follow it.